net/nginx/files/nginx.init

```diff
@@ -5,13 +5,41 @@ START=80
 
 USE_PROCD=1
 
+NGINX_UTIL="/usr/bin/nginx-util"
+
+eval $("${NGINX_UTIL}" get_env)
+
 start_service() {
 	[ -d /var/log/nginx ] || mkdir -p /var/log/nginx
 	[ -d /var/lib/nginx ] || mkdir -p /var/lib/nginx
 
+	${NGINX_UTIL} init_lan
+
 	procd_open_instance
-	procd_set_param command /usr/sbin/nginx -c /etc/nginx/nginx.conf -g 'daemon off;'
-	procd_set_param file /etc/nginx/nginx.conf
+	NCPUS="$(grep -c '^processor\s*:' /proc/cpuinfo)"
+	procd_set_param command /usr/sbin/nginx -c "${NGINX_CONF}" \
+		-g "daemon off; worker_processes $NCPUS;"
+	procd_set_param stdout 1;
+	procd_set_param stderr 1;
+	procd_set_param file "${LAN_LISTEN}" "${LAN_LISTEN}.default" \
+		"${NGINX_CONF}" "${CONF_DIR}*.conf" "${CONF_DIR}*.locations"
+	if [ "${LAN_SSL_LISTEN}" != "" ]
+	then
+		procd_append_param file "${CONF_DIR}*.crt" "${CONF_DIR}*.key" \
+			"${LAN_SSL_LISTEN}" "${LAN_SSL_LISTEN}.default"
+	fi
 	procd_set_param respawn
 	procd_close_instance
 }
+
+stop_service() {
+	rm -f "${LAN_LISTEN}" "${LAN_LISTEN}.default"
+	if [ "${LAN_SSL_LISTEN}" != "" ]
+	then rm -f "${LAN_SSL_LISTEN}" "${LAN_SSL_LISTEN}.default"
+	fi
+}
+
+service_triggers() {
+	procd_add_reload_interface_trigger loopback
+	procd_add_reload_interface_trigger lan
+}
```

Comment:
Would reload_service be applicable here? (sending SIGHUP instead of service stop && service start). I have no idea if nginx has support for that.

Reply:
Yes, it is. Thank you. 
There is also a SIGUSR1 for reopening the log files, added that as EXTRA :-)